runbot/runbot.py

```diff
@@ -1243,11 +1244,27 @@ def branch_info(branch):
 
         return request.render("runbot.repo", context)
 
-    @http.route(['/runbot/hook/<int:repo_id>'], type='http', auth="public", website=True)
+    @http.route(['/runbot/hook/<int:repo_id>', '/runbot/hook/org'], type='json', auth="public", website=True)
     def hook(self, repo_id=None, **post):
-        # TODO if repo_id == None parse the json['repository']['ssh_url'] and find the right repo
-        repo = request.registry['runbot.repo'].browse(request.cr, SUPERUSER_ID, [repo_id])
-        repo.hook_time = datetime.datetime.now().strftime(openerp.tools.DEFAULT_SERVER_DATETIME_FORMAT)
+        if repo_id is None:
+            repo_data = request.jsonrequest.get('repository')
+            event = request.httprequest.headers.get("X-Github-Event")
+            if repo_data and event in ['push', 'pull_request']:
+                repo_owner = repo_data['owner'].get('name') or repo_data['owner'].get('login')
+                repo_name = repo_data['name']
+                repo_domain = [
+                    '|', ('name', '=', 'git@github.com:%s/%s.git' % (repo_owner, repo_name)),
+                    ('name', '=', 'https://github.com/%s/%s.git' % (repo_owner, repo_name)),
```

Comment:
And actually, instead of harcoding the URL based on the owner/repo name, why don't you directly use `repo.ssh_url` and `repo.git_url` from the json data?
Please keep in mind that the `.git` suffix is entirely optional, so it should work with or without it. Github seems to always include the suffix in both `repo.ssh_url` and `repo.git_url`. See [the doc](https://developer.github.com/v3/activity/events/types/#repositoryevent).

This makes the hook much less github-specific, as other repositories and hooks can now use it, as long as they provide a proper `repo.ssh_url` or `repo.git_url` in the json data.